gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/QueryResultsImpl.java

```diff
@@ -28,46 +28,49 @@
 
   private final DatastoreImpl datastore;
-  private final DatastoreV1.ReadOptions readOptionsPb;
-  private final DatastoreV1.PartitionId partitionIdPb;
+  private final com.google.datastore.v1beta3.ReadOptions readOptionsPb;
+  private final com.google.datastore.v1beta3.PartitionId partitionIdPb;
   private final ResultType<T> queryResultType;
   private Query<T> query;
   private ResultType<?> actualResultType;
-  private DatastoreV1.QueryResultBatch queryResultBatchPb;
+  private com.google.datastore.v1beta3.QueryResultBatch queryResultBatchPb;
   private boolean lastBatch;
-  private Iterator<DatastoreV1.EntityResult> entityResultPbIter;
-  //private ByteString cursor; // only available in v1beta3
+  private Iterator<com.google.datastore.v1beta3.EntityResult> entityResultPbIter;
+  private ByteString cursor;
 
 
-  QueryResultsImpl(DatastoreImpl datastore, DatastoreV1.ReadOptions readOptionsPb,
+  QueryResultsImpl(DatastoreImpl datastore, com.google.datastore.v1beta3.ReadOptions readOptionsPb,
                    Query<T> query) {
     this.datastore = datastore;
     this.readOptionsPb = readOptionsPb;
     this.query = query;
     queryResultType = query.type();
-    DatastoreV1.PartitionId.Builder pbBuilder = DatastoreV1.PartitionId.newBuilder();
-    pbBuilder.setDatasetId(datastore.options().projectId());
+    com.google.datastore.v1beta3.PartitionId.Builder pbBuilder =
+        com.google.datastore.v1beta3.PartitionId.newBuilder();
+    pbBuilder.setProjectId(datastore.options().projectId());
     if (query.namespace() != null) {
-      pbBuilder.setNamespace(query.namespace());
+      pbBuilder.setNamespaceId(query.namespace());
     } else if (datastore.options().namespace() != null) {
-      pbBuilder.setNamespace(datastore.options().namespace());
+      pbBuilder.setNamespaceId(datastore.options().namespace());
     }
     partitionIdPb = pbBuilder.build();
     sendRequest();
   }
 
   private void sendRequest() {
-    DatastoreV1.RunQueryRequest.Builder requestPb = DatastoreV1.RunQueryRequest.newBuilder();
+    com.google.datastore.v1beta3.RunQueryRequest.Builder requestPb =
+        com.google.datastore.v1beta3.RunQueryRequest.newBuilder();
     if (readOptionsPb != null) {
       requestPb.setReadOptions(readOptionsPb);
     }
     requestPb.setPartitionId(partitionIdPb);
     query.populatePb(requestPb);
-    // TODO(ajaykannan): fix me!
-    //queryResultBatchPb = datastore.runQuery(requestPb.build()).getBatch();
+    queryResultBatchPb = datastore.runQuery(requestPb.build()).getBatch();
     lastBatch = queryResultBatchPb.getMoreResults() != MoreResultsType.NOT_FINISHED;
-    entityResultPbIter = queryResultBatchPb.getEntityResultList().iterator();
-    // cursor = resultPb.getSkippedCursor(); // available in v1beta3, use startCursor if not skipped
+    entityResultPbIter = queryResultBatchPb.getEntityResultsList().iterator();
+    if (queryResultBatchPb.getSkippedResults() > 0) {
```

Comment:
I think `startCursor` still exists in `com.google.datastore.v1beta3.Query`, but I don't think it ever existed in `QueryResultsBatch`.